sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/MethodParameterLookupTest.cs

```diff
@@ -0,0 +1,290 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2019 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.UnitTest.TestFramework;
+using CSharpSyntax = Microsoft.CodeAnalysis.CSharp.Syntax;
+using VBSyntax = Microsoft.CodeAnalysis.VisualBasic.Syntax;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class MethodParameterLookupTest
+    {
+
+
+        [TestMethod]
+        public void TestMethodParameterLookup_CS()
+        {
+            const string Source = @"
+namespace Test
+{
+    class TestClass
+    {
+        void Main()
+        {
+            DoNothing();
+            DoSomething(1, true);
+            DoSomething(b: true, a: 1);
+            WithOptional(1);
+            WithOptional(1, ""Ipsum"");
+            WithOptional(opt: ""Ipsum"", a: 1);
+            WithParams();
+            WithParams(1, 2, 3);
+        }
+
+        void ThisShouldNotBeFoundInMain()
+        {
+            SpecialMethod(65535);
+        }
+
+        void DoNothing()
+        {
+        }
+
+        void DoSomething(int a, bool b)
+        {
+        }
+
+        void WithOptional(int a, string opt = ""Lorem"")
+        {
+        }
+
+        void WithParams(params int[] arr)
+        {
+        }
+
+        void SpecialMethod(int specialParameter)
+        {
+        }
+    }
+}";
+            var c = new CSharpInspection(Source);
+            c.Inspect(0, "DoNothing", new { });
```

Comment:
I would suggest to rename the `Inspect` method to something more explicit, like `HasExpectedParameterMappings(...` to be clear at first glance on what it does